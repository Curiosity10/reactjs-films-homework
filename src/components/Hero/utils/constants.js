export const MAX_OVERVIEW_LENGTH = 220;
export const DEFAULT_MOVIE_RUNTIME = '0h 0m';
