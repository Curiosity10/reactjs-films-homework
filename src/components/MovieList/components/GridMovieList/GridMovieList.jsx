import React, { useState } from 'react';
import GridMovieCard from './components/GridMovieCard';

const GridMovieList = ({ movies }) => {
  const [activeCard, setActiveCard] = useState(0);
  return movies.map(movie => (
    <GridMovieCard
      activeCard={activeCard}
      changeActiveCard={id => setActiveCard(id)}
      key={movie.id}
      movie={movie}
    />
  ));
};

export default GridMovieList;
