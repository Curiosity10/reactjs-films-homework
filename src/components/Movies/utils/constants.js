export const MAX_OVERLAY_CARD_DESC_LENGTH = 180;
export const MAX_TABLE_CARD_DESC_LENGTH = 500;
export const MAX_SCROLL_HEIGHT = 0.99;
