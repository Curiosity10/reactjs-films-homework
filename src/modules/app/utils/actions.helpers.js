export const createRequestAction = requestName => ({
  type: requestName,
});

export const createFailureAction = (requestName, error) => ({
  type: requestName,
  payload: {
    error: error.message,
  },
});

export const createReceiveMoviesAction = ({
  type, data, page, genre = null, query = '',
}) => ({
  type,
  payload: {
    movies: data.results,
    currentPage: page,
    pagesCount: data.total_pages,
    genreId: genre,
    searchQuery: query,
  },
});

export const createAsyncAction = ({
  type, asyncFn, page = 1, genre = null, query = '', filterType = '',
}) => (dispatch) => {
  dispatch(createRequestAction(`${type}_REQUEST`));
  return asyncFn({ page, genre, query }, filterType)
    .then(response => response.json())
    .then(data => dispatch(createReceiveMoviesAction({
      type: `${type}_SUCCESS`, data, page, genre, query,
    })))
    .catch(error => dispatch(createFailureAction(`${type}_FAILURE`, error)));
};
