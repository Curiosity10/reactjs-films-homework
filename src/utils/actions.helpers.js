export const createAsyncFnRequest = requestName => ({
  type: requestName,
});

export const createAsyncFnFailure = (requestName, error) => ({
  type: requestName,
  payload: {
    error: error.message,
  },
});

const createReceiveMovies = (type, data, page) => ({
  type,
  payload: {
    movies: data.results,
    currentPage: page,
    pagesCount: data.total_pages,
  },
});

export const createAsyncFnSuccess = (type, asyncFn, page = 1) => (dispatch) => {
  dispatch(createAsyncFnRequest(`${type}_REQUEST`));
  return asyncFn({ page })
    .then(response => response.json())
    .then(data => dispatch(createReceiveMovies(`${type}_SUCCESS`, data, page)))
    .catch(error => dispatch(createAsyncFnFailure(`${type}_FAILURE`, error)));
};
