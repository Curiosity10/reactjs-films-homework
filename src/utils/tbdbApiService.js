import Movie from '../assets/json/movie.json';
import Genres from '../assets/json/genres.json';
import Popular from '../assets/json/popular.json';

const urlBase = 'https://image.tmdb.org/t/p/original';
const urlMinImg = 'https://image.tmdb.org/t/p/w300';

function getNormalRuntime(time) {
  const hours = Math.floor(time / 60);
  const minutes = time - 60 * hours;
  return `${hours}h ${minutes}m`;
}

function getMovieGenres(genresIds) {
  const genresArr = [];
  genresIds.map(genre => Genres.genres.map((item) => {
    if (item.id === genre) {
      return genresArr.push({ id: item.id, name: item.name });
    }
    return 0;
  }));
  return genresArr;
}

export function getMovieInfo() {
  return {
    title: Movie.original_title,
    genres: Movie.genres,
    rating: Movie.vote_average,
    runtime: getNormalRuntime(Movie.runtime),
    overview: Movie.overview,
    image: `${urlBase}${Movie.backdrop_path}`,
  };
}

export function getGenres() {
  return Genres.genres;
}

export function getPopularMovies() {
  return Popular.results.map(movie => ({
    id: movie.id,
    rating: movie.vote_average,
    title: movie.title,
    genres: getMovieGenres(movie.genre_ids),
    overview: movie.overview,
    image: `${urlMinImg}${movie.poster_path}`,
  }));
}
