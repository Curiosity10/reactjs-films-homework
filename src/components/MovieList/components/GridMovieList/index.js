import GridMovieList from './GridMovieList';

export default GridMovieList;
