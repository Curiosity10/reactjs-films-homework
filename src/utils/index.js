import {
  getTopMovies,
  searchMovies,
  getGenres,
  getMoviesByGenre,
  getLatestMovies,
  getUpcomingMovies,
  getMovieVideo,
} from './api';
import { getMovies } from './tbdbApiService';
import truncate from './truncate';
import {
  createFailureAction,
  createRequestAction,
  createAsyncAction,
  createReceiveMoviesAction,
} from './actions.helpers';

export {
  getTopMovies,
  searchMovies,
  getGenres,
  getMoviesByGenre,
  getLatestMovies,
  getUpcomingMovies,
  getMovieVideo,
  truncate,
  getMovies,
  createFailureAction,
  createRequestAction,
  createAsyncAction,
  createReceiveMoviesAction,
};
