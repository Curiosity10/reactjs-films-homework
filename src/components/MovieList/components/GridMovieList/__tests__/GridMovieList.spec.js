import React from 'react';
import { act, create } from 'react-test-renderer';
import GridMovieList from '..';
import { getMovieInfo } from '../../../../../utils/tbdbApiService';

describe('GridMovieList component renders correctly', () => {
  const movies = [{ ...getMovieInfo(), id: 2 }];

  it('GridMovieList renders correctly', () => {
    const tree = create(<GridMovieList
      movies={movies}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Change active card works correctly', () => {
    const tree = create(<GridMovieList movies={movies} />);
    const mockFunction = jest.fn(tree.toTree().rendered.props.changeActiveCard);
    act(() => {
      mockFunction();
    });
    expect(mockFunction).toHaveBeenCalled();
  });
});
