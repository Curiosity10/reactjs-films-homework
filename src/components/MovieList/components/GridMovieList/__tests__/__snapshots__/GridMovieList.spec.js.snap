// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridMovieList component renders correctly GridMovieList renders correctly 1`] = `
<article
  className="card"
>
  <img
    alt="Poster"
    height="75%"
    src="https://image.tmdb.org/t/p/original/59qyrVGLLtrcuFtJGdixzn7H4OI.jpg"
  />
  <div
    className="onHover"
  >
    <button
      aria-label="Watch"
      className="btnWatch"
      type="button"
    >
      <svg
        height="80px"
        viewBox="0 0 58 58"
        width="80px"
        x="0px"
        y="0px"
      >
        <circle
          cx="29"
          cy="29"
          r="29"
          style={
            Object {
              "fill": "#0aaee4",
            }
          }
        />
        <g>
          <polygon
            points="44,29 22,44 22,29.273 22,14"
            style={
              Object {
                "fill": "#fff",
              }
            }
          />
          <path
            d="M22,45c-0.16,0-0.321-0.038-0.467-0.116C21.205,44.711,21,44.371,21,44V14 c0-0.371,0.205-0.711,0.533-0.884c0.328-0.174,0.724-0.15,1.031,0.058l22,15C44.836,28.36,45,28.669,45,29s-0.164,0.64-0.437,0.826 l-22,15C22.394,44.941,22.197,45,22,45z M23,15.893v26.215L42.225,29L23,15.893z"
            style={
              Object {
                "fill": "#fff",
              }
            }
          />
        </g>
      </svg>
      <span>
         Watch Now 
      </span>
    </button>
    <button
      aria-label="View Info"
      className="btn transparentBtn"
      onClick={[Function]}
      type="button"
    >
      View Info
    </button>
  </div>
  <div
    className="info"
  >
    <div>
      <h3
        className="info_title"
      >
        Ð¢-34
      </h3>
      <p>
        <span
          className="genre"
          style={
            Object {
              "color": "#0aaee4",
              "fontSize": "0.8rem",
            }
          }
        >
          War
        </span>
        <span
          className="genre"
          style={
            Object {
              "color": "#0aaee4",
              "fontSize": "0.8rem",
            }
          }
        >
          Drama
        </span>
        <span
          className="genre"
          style={
            Object {
              "color": "#0aaee4",
              "fontSize": "0.8rem",
            }
          }
        >
          Adventure
        </span>
        <span
          className="genre"
          style={
            Object {
              "color": "#0aaee4",
              "fontSize": "0.8rem",
            }
          }
        >
          Action
        </span>
      </p>
    </div>
    <span
      className="numRating"
      style={
        Object {
          "color": "#0aaee4",
        }
      }
    >
      4.7
    </span>
  </div>
</article>
`;
